.github/workflows/tests.yml

```diff
@@ -0,0 +1,117 @@
+name: Tests
+
+permissions: {}
+
+on:
+  push:
+    branches: ["**"]
+  pull_request:
+    branches: ["**"]
+
+jobs:
+  tests:
+    name: Tests
+    permissions:
+      contents: read
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.13", "3.12", "3.11", "3.10", "3.9", "3.8"]
+    steps:
+      - name: Install prerequisites
+        run: |
+          sudo apt-get update
+          sudo apt-get install libboost-all-dev libgsl0-dev libeigen3-dev
+
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
+
+      - name: Build and Install
+        run: |
+          cd similarity_search
+          cmake . -DWITH_EXTRAS=1
+          make
+          cd ..
+
+      - name: Test with pytest and generate coverage file
+        run:
+          cd python_bindings && uvx -p ${{ matrix.python-version }} --with
+          tox-uv tox -e py
+
+  build_wheels:
+    name: Build
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ["3.13", "3.12", "3.11", "3.10", "3.9", "3.8"]
+        os:
+          - ubuntu-latest
+          - ubuntu-24.04-arm
+          - windows-latest
+          # - windows-11-arm # arm is busted, CIBW_ARCHS_WINDOWS=x86_64
```

Comment:
We have a separate build for Windows. It is only for Intel, but Windows arm is uncommon, so I wouldn't bother.